app/peptide/payloadstore/payloadstore.go

```diff
@@ -1,70 +1,100 @@
 package payloadstore
 
 import (
-	"sync"
+	"fmt"
 
 	"github.com/ethereum/go-ethereum/beacon/engine"
 	"github.com/polymerdao/monomer"
 )
 
-type PayloadStore interface {
-	Add(payload *monomer.Payload)
-	Get(id engine.PayloadID) (*monomer.Payload, bool)
-	Current() *monomer.Payload
-	RollbackToHeight(height int64) error
-}
-
-type pstore struct {
-	mutex    sync.Mutex
+type Store struct {
```

Comment:
Let's add a comment noting that `Store` is not goroutine-safe.

Reply:
45518ed